lib/advanced-open-file-view.coffee

```diff
@@ -167,6 +176,13 @@ class AdvancedFileView extends View
     @miniEditor.setText newPath
 
   update: ->
+    if atom.config.get 'advanced-open-file.helmDirSwitch'
+      text = @miniEditor.getText()
+      if text.endsWith(path.sep + path.sep)
+        @updatePath @FS_ROOT
+      else if text.endsWith("~#{path.sep}")
```

Comment:
I think we might want to check for `/~/` here, because directories could easily end with `~`, and putting a / after those shouldn't switch to the home directory.